binary_search_tree/tree.py

```diff
@@ -7,45 +7,128 @@ def __init__(self, key, val = None):
+                return None
+            return self.find_helper(current_root.left, key)
+
+        if key > current_root.key:
+            if current_root.right is None:
+                return None
+            return self.find_helper(current_root.right, key)
+
+    # Time Complexity: O(log n)
+    # Space Complexity: O(1)
     def find(self, key):
-        pass
+        if self.root is None:
+            return None
+
+        return self.find_helper(self.root, key)
 
-    # Time Complexity: 
-    # Space Complexity: 
+    def inorder_helper(self, current_node, items):
+
+        if current_node is not None:
+
+            self.inorder_helper(current_node.left, items)
+            items.append({"key": current_node.key, "value": current_node.value})
+            self.inorder_helper(current_node.right, items)
+
+    # This method returns an array of all the elements in the tree, in order
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def inorder(self):
-        pass
+        items = []
+        self.inorder_helper(self.root,items)
+        return items
 
-    # Time Complexity: 
-    # Space Complexity:     
+    def preorder_helper(self, current_node, items):
+
+        if current_node is None:
+            return
+        
+        items.append({"key": current_node.key, "value": current_node.value})
+
+        self.preorder_helper(current_node.left, items)
+        self.preorder_helper(current_node.right, items)
+
+    # This method returns an array of all the elements in a preorder fashion (root, left, right)
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def preorder(self):
-        pass
+        items = []
+        self.preorder_helper(self.root, items)
+        return items
+
+    def postorder_helper(self, current_node, items):
+        if current_node is None:
+            return
+        
+        self.postorder_helper(current_node.left, items)
+        self.postorder_helper(current_node.right, items)
+
 
-    # Time Complexity: 
-    # Space Complexity:     
+
+        items.append({"key": current_node.key, "value": current_node.value})
+
+    # This method returns an array of all the elements in a postorder fashion (left, right , root)
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def postorder(self):
-        pass
+        items = []
+        self.postorder_helper(self.root, items)
+        return items
 
-    # Time Complexity: 
-    # Space Complexity:     
-    def height(self):
-        pass
+    def height_helper(self, current_node):
+        if current_node is not None:
+            height_left = self.height_helper(current_node.left)
+            height_right = self.height_helper(current_node.right)
+            return max(height_left, height_right) + 1
+        
+        else:
+            return 0
 
 
+    # This method returns the height of the binary search tree
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
+    def height(self):
```

Comment:
✨ Yes! Time and space complexity would be O(n) if the tree is unbalanced. If the tree is balanced, time complexity would be O(log n)